### PR TITLE
Moving Transform: Tests and fixes

### LINKED ISSUE #204: Moving Transform: mop up
#197 was merged because it was high time to release a new version.

- Moving Transform works better than it used to, but still requires tests (it had none) and new documentation.
- Remove Aggregate widget, add `replaces` to Moving Transform.

### PR DESCRIPTION
**Pull request by janezd**
##### Issue

#197 was merged without tests. (Not that the widget had any previously. :)

Fixes #204.

##### Description of changes

Add tests, fix detected problems.

##### Includes
- [X] Code changes
- [X] Tests- [ ] Documentation

### CODE REVIEW

File: orangecontrib/timeseries/widgets/owmovingtransform.py [hunk 1]
```diff
@@ -205,6 +206,11 @@ class Warning(widget.OWWidget.Warning):
         window_to_large = widget.Msg("Window width is to large")
```
Comment (ajdapretnar) [inline orangecontrib/timeseries/widgets/owmovingtransform.py:206]: I didn't notice this before, but it should be "Window width is too large."
Same below.

Comment (janezd) [reply orangecontrib/timeseries/widgets/owmovingtransform.py]: 👍

File: doc/widgets/moving_transform_w.md [hunk 2]
```diff
@@ -13,26 +13,51 @@ Apply rolling window functions to the time series. Use this widget to get a seri
...
-1. Define a new transformation.
-2. Remove the selected transformation.
-3. Time series you want to run the transformation over.
-4. Desired window size.
-5. Aggregation function to aggregate the values in the window with. Options are: *mean*, *sum*, *max*, *min*, *median*, *mode*, *standard deviation*, *variance*, *product*, *linearly-weighted moving average*, *exponential moving average*, *harmonic mean*, *geometric mean*, *non-zero count*, *cumulative sum*, and *cumulative product*.
-6. Select *Non-overlapping windows* options if you don't want the moving windows to overlap but instead be placed side-to-side with zero intersection.
-7. In the case of non-overlapping windows, define the fixed window width(overrides and widths set in (4).
+1. Define the aggregation type:
+   - *Sliding window*: data will be aggregated with the sliding window of the specified width. The function is typically used for smoothing the data.
+      - *Window width*: the number of instances used for creating aggregation
```
Comment (janezd) [inline doc/widgets/moving_transform_w.md:20]: `the number of instances in a window`?

File: doc/widgets/moving_transform_w.md [hunk 3]
```diff
@@ -13,26 +13,51 @@ Apply rolling window functions to the time series. Use this widget to get a seri
...
+      - define data output: *Discard original data* will output only aggregated columns, *Keep original data* will output both the original columns and the aggregated columns, *Include leading instances* will output original and aggregated column, including the leading instances that fall outside of the window
```
Comment (janezd) [inline doc/widgets/moving_transform_w.md:21]: "Falling outside the window" sounds as if these are the last instances.

Maybe:

*Include leading instances* will output original and aggregated column, including the first N - 1 leading instances (where N is the window width) before the first complete window. The values in aggregated columns for these instances are left undefined.

File: doc/widgets/moving_transform_w.md [hunk 4]
```diff
@@ -13,26 +13,51 @@ Apply rolling window functions to the time series. Use this widget to get a seri
...
+   - *Consecutive blocks*: data will be aggregated using disjunctive blocks instead of sliding widows (instances won't overlap).
```
Comment (janezd) [inline doc/widgets/moving_transform_w.md:22]: This is not really a sliding window, and mentioning overlaps is imho not necessary.

Maybe:

data will be split into consecutive blocks of the given size

File: doc/widgets/moving_transform_w.md [hunk 5]
```diff
@@ -13,26 +13,51 @@ Apply rolling window functions to the time series. Use this widget to get a seri
...
+      - *Block width*: the number of instances used for creating aggregation
+      - define data output: *Discard original data* will output only aggregated columns, *Keep first instance* will output data with only the first instances from the blocks included, *Keep middle instance* will output data with only the middle instances from the blocks included, *Keep last instance* will output data with only the last instances from the blocks included
```
Comment (janezd) [inline doc/widgets/moving_transform_w.md:24]: *Keep first instance*, *Keep middle instance* and *Keep last instance* will take the first, middle or last data instance as the representative for the block.

File: doc/widgets/moving_transform_w.md [hunk 6]
```diff
@@ -13,26 +13,51 @@ Apply rolling window functions to the time series. Use this widget to get a seri
...
+   - *Aggregate time periods*: data will be aggregated with the "group-by" principle. The function replace the deprecated **Aggregate** widget. The aggregation will aggregated the data using the provided time period. Possible periods are years, months, days, hours, minutes, seconds, month of year (12 instances), day of year (365 instances), day of month (~31 instances), day of week (7 instances), hour of day (24 instances).
```
Comment (janezd) [inline doc/widgets/moving_transform_w.md:25]: What is "group-by" principle? I'd remove the first sentence and put the last to the end.

- *Aggregate time periods*: will aggregate data based on time periods (years, months, days, hours, minutes, seconds) or in the same month of year (12 instances), day of year (365 or 366 instances), day of month (~31 instances), day of week (7 instances), hour of day (24 instances). This functionality replaces the **Aggregate** widget.

File: doc/widgets/moving_transform_w.md [hunk 7]
```diff
@@ -13,26 +13,51 @@ Apply rolling window functions to the time series. Use this widget to get a seri
...
+2. If *Apply Automatically* is ticked, changes are communicated automatically. Alternatively, click *Apply*.
+3. Variable selection:
+   - Filter provides a shortcut for listing desired variables. Start typing the variable name to select it from the list.
```
Comment (janezd) [inline doc/widgets/moving_transform_w.md:28]: - Filter provides a shortcut for searching variables by (a part of) their name. Start typing ...

File: doc/widgets/moving_transform_w.md [hunk 8]
```diff
@@ -13,26 +13,51 @@ Apply rolling window functions to the time series. Use this widget to get a seri
...
+   - Alternatively, select the variable from the variable list. You can use Ctrl (Cmd) command to select multiple variables and apply the same aggregation on them.
```
Comment (janezd) [inline doc/widgets/moving_transform_w.md:29]: command to select -> key, or drag over the list to select

File: doc/widgets/moving_transform_w.md [hunk 9]
```diff
@@ -13,26 +13,51 @@ Apply rolling window functions to the time series. Use this widget to get a seri
...
+   - *Show only numeric variables* removes non-numeric variables from the variable list.
```
Comment (janezd) [inline doc/widgets/moving_transform_w.md:30]: removes -> hides?

**Review by janezd — COMMENTED** [review]
A few comments. After you're done, you can merge (if my commits are OK).

**Pull request merged**